build.gradle

```diff
@@ -271,6 +282,40 @@ project(':sbe-tool') {
 
     compileTestJava.dependsOn compileGeneratedJava
 
+    testing {
+        suites {
+            test {
+                dependencies {
+                    implementation files('build/classes/java/generated')
+                    implementation "org.hamcrest:hamcrest:${hamcrestVersion}"
+                    implementation "org.mockito:mockito-core:${mockitoVersion}"
+                    implementation "org.junit.jupiter:junit-jupiter-params:${junitVersion}"
+                }
+            }
+
+            propertyTest(JvmTestSuite) {
+                useJUnitJupiter junitVersion
+
+                dependencies {
+                    implementation project()
+                    implementation "net.jqwik:jqwik:${jqwikVersion}"
+                    implementation "org.json:json:${jsonVersion}"
```

Comment:
Note that we're using `jvm-test-suites` to add new source sets in a Gradle 9 compatible manner.
Here `implementation` means `propertyTestImplementation` in old money.